lib/matplotlib/collections.py

```diff
@@ -1254,6 +1255,285 @@ def set_verts_and_codes(self, verts, codes):
+            raise ValueError(msg)
+
+    def set_data(
+            self, t, f1, f2, *,
+            where=None, interpolate=False, step=None, **kwargs):
+        """
+        Set the two curves and update the collection.
+
+        Parameters
+        ----------
+        t : array (length N)
+            The ``self.t_direction`` coordinates of the nodes defining the curves.
+
+        f1 : array (length N) or scalar
+            The other coordinates of the nodes defining the first curve.
+
+        f2 : array (length N) or scalar
+            The other coordinates of the nodes defining the second curve.
+
+        where : array of bool (length N), optional
+            Define *where* to exclude some {dir} regions from being filled.
+            The filled regions are defined by the coordinates ``t[where]``.
+            More precisely, fill between ``t[i]`` and ``t[i+1]`` if
+            ``where[i] and where[i+1]``.  Note that this definition implies
+            that an isolated *True* value between two *False* values in *where*
+            will not result in filling.  Both sides of the *True* position
+            remain unfilled due to the adjacent *False* values.
+
+        interpolate : bool, default: False
+            This option is only relevant if *where* is used and the two curves
+            are crossing each other.
+
+            Semantically, *where* is often used for *f1* > *f2* or
+            similar.  By default, the nodes of the polygon defining the filled
+            region will only be placed at the positions in the *t* array.
+            Such a polygon cannot describe the above semantics close to the
+            intersection.  The t-sections containing the intersection are
+            simply clipped.
+
+            Setting *interpolate* to *True* will calculate the actual
+            intersection point and extend the filled region up to this point.
+
+        step : {{'pre', 'post', 'mid'}}, optional
+            Define *step* if the filling should be a step function,
+            i.e. constant in between *t*.  The value determines where the
+            step will occur:
+
+            - 'pre': The f value is continued constantly to the left from
+              every *t* position, i.e. the interval ``(t[i-1], t[i]]`` has the
+              value ``f[i]``.
+            - 'post': The y value is continued constantly to the right from
+              every *x* position, i.e. the interval ``[t[i], t[i+1])`` has the
+              value ``f[i]``.
+            - 'mid': Steps occur half-way between the *t* positions.
+
+        **kwargs
+            Forwarded to `.PolyCollection`.
+
+        See Also
+        --------
+        .PolyCollection.set_verts, .Line2D.set_data
+        """
+        kwargs = self._normalize_kwargs(self.axes, **kwargs)
+        polys, _ = self._make_verts_and_pts(
+            t, f1, f2, where, interpolate, step, self.axes, **kwargs)
```

Comment:
```suggestion
            t, f1, f2, where, interpolate, step, self.axes, **kwargs)
```
The only reason axes and kwargs are passed on here is giving them to `axes._process_unit_info()`.

I'm not an expert in unit handling, but suspect that units should be handled outside of the artists. This would likely imply that we don't have to bother with it in `set_data` at all (i.e. remove `axes` and `kwargs` from `_make_verts_and_pts` and `set_data`. The initial unit handling `axes._process_unit_info()` (currently via `__init__` -> `_make_verts_and_pts`) could then move up into `_fill_between_x_or_y`.

@anntzer do you agree?

This would further untangle the `axes` dependency.

Reply:
Hi, is there a typo in the `suggested change` section? Currently I see no changes.

Reply:
Whoops. I think I intended to remove `self.axes` and `**kwargs` as a suggestion, but then realized that the change requires updates in multiple places, so a simple suggestion is not helpful. -> I've removed the code suggestion. But the idea of removing the parameters still stands.

Reply:
Good catch regarding unit handling.  Indeed nearly all artists only handle unitless data, and the de-unitization should be done at the level of the Axes method.  (The only exception is Line2D, which has fairly complicated ad-hoc machinery to handle units (including change of units) itself, and which I think we should not try to replicate at least until the data prototype potentially gives us a more comprehensive way to handle this across all artists.)

Reply:
After 84c5e559a8a6d6c713bb1a5684ec19d7de40d20e, `axes` is no more an argument.